v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/pubsubtotext/PubsubToTextIT.java

```diff
@@ -0,0 +1,243 @@
+
+  @BeforeClass
+  public static void setUpClass() throws IOException {
+    Storage gcsClient = createGcsClient(CREDENTIALS);
+
+    TopicAdminClient topicAdminClient =
+        TopicAdminClient.create(
+            TopicAdminSettings.newBuilder().setCredentialsProvider(CREDENTIALS_PROVIDER).build());
+
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(PROJECT).topicAdminClient(topicAdminClient).build();
+
+    artifactClient = GcsArtifactClient.builder(gcsClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    artifactClient.cleanupRun();
+    pubsubResourceManager.cleanupAll();
+  }
+
+  @Test
+  public void testTopicToGcs() throws IOException {
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    String messageString = String.format("msg-%s", jobName);
+    Pattern expectedFilePattern = Pattern.compile(".*topic-output-.*");
+
+    Topic topic = pubsubResourceManager.createTopic(jobName);
+    LaunchConfig options =
+        LaunchConfig.builder(jobName, SPEC_PATH)
+            .addParameter(INPUT_TOPIC, topic.getName())
+            .addParameter(WINDOW_DURATION_KEY, DEFAULT_WINDOW_DURATION)
+            .addParameter(OUTPUT_DIRECTORY_KEY, getTestMethodDirPath(name))
+            .addParameter(NUM_SHARDS_KEY, "1")
+            .addParameter(OUTPUT_FILENAME_PREFIX, "topic-output-")
+            .build();
+    DataflowTemplateClient dataflow =
+        FlexTemplateClient.builder().setCredentials(CREDENTIALS).build();
+
+    // Act
+    JobInfo info = dataflow.launchTemplate(PROJECT, REGION, options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+
+    PubsubPublisher publisher =
+        DefaultPubsubPublisher.builder(topic.getName())
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    AtomicReference<List<Artifact>> artifacts = new AtomicReference<>();
+
+    Result result =
+        new DataflowOperator(dataflow)
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () -> {
+                  try {
+                    ByteString messageData = ByteString.copyFromUtf8(messageString);
+                    publisher.publish(ImmutableMap.of(), messageData);
+                  } catch (ExecutionException | InterruptedException e) {
+                    throw new RuntimeException("Error publishing message to Pub/Sub topic", e);
+                  }
+
+                  artifacts.set(artifactClient.listArtifacts(name, expectedFilePattern));
+                  return !artifacts.get().isEmpty();
+                });
+
+    // Assert
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
+
+    // Make sure that files contain the job name
+    assertThat(
+        artifacts.get().stream()
+                .anyMatch(artifact -> new String(artifact.contents()).contains(messageString)))
+        .isTrue();
```

Comment:
Valid concern. I am saving all the contents to a single stream and `assertThat(...).contains()` now. Do you think that is enough? Our PubsubMessage's data is only sending `messageString`, so we expect at least that `messageString` is on the files. 

Any ideas?

Reply:
I guess I'm wondering if (or why) there'd be anything other than `messageString`. Is this expected behavior, or does it indicate a problem with the pipeline? If the former, this is fine. If the latter, I'm a bit concerned that this would hide an error with the pipeline.

Reply:
Yeah, but having unexpected data is also a bug, and since we only pass a single message to Pub/Sub, I would think we should know the exact data and its order (order isn't possible to know on multiple inputs). That said, I'd just prefer trying to be more specific if possible. If that's not feasible, feel free to just resolve this comment.

Reply:
Got it. I've changed slightly to replace anything that we expect for "", and validate if it yields empty. It seems a bit more convoluted, but I agree it's better / covers the no unexpected data case.